lib/spack/spack/util/compression.py

```diff
@@ -168,65 +168,57 @@ def _unzip(archive_file):
 
 
 def _unZ(archive_file):
-    if sys.platform == "win32":
-        result = _7zip(archive_file)
-    else:
-        result = _system_gunzip(archive_file)
-    return result
+    return _system_gunzip(archive_file)
```

Comment:
Out of curiosity, why remove support for using 7zip? With the other changes (e.g. to `_win_compressed_tarball_handler`) we will default to using built-in Python support, but for `.Z` archives would it be problematic to fall back on `_7zip`?

Reply:
In general we should be trying to keep the number of dependencies needed to just get Spack running to a minimum, but I think in this case there's really no other option as there are no python options to extract `Z` files, and so long as we can install clingo, we can install 7zip with Spack, so I think it can be returned here. 
I know Spack loads all externally detected package prefixes into the `PATH` but is the same true with installed packages? Or would we need to manually make a spack installed 7zip available to the `which` method?

Reply:
> In general we should be trying to keep the number of dependencies needed to just get Spack running to a minimum,

Agree. But to be clear I don't think that the other changes in the PR already achieve that goal. For packages which happen to distribute `.Z`, providing the `_7zip` package would allow Spack to decompress them if the user installs 7zip.

> I know Spack loads all externally detected package prefixes into the PATH but is the same true with installed packages? 

That isn't automatic: we'd have to update our bootstrapping logic to also search for and include 7zip on Windows (but that could be done in some future PR, just allowing users to build `.Z`-based packages for now if they manually install 7zip is sufficient).

Reply:
This PR does not stop providing the the `_7zip` package from spack to users, it simply removes the dependence of the compression module on 7zip. Prior to this PR, it is impossible to use Spack without 7zip, as 7zip was required to extract the clingo compressed archive, which makes it a fundamental dependency, something we should avoid. 
This goal of this PR was to get 7zip to a place like `cmake` where we don't rely on it to get started but can install via Spack and use it from there. 

> That isn't automatic: we'd have to update our bootstrapping logic to also search for and include 7zip on Windows (but that could be done in some future PR, just allowing users to build .Z-based packages for now if they manually install 7zip is sufficient).

I had intended to mean loading a package installed via `spack install 7zip` rather than one that is bootstrapped. Agreed 7zip should probably be included in the bootstrap in a separate PR, but I'd strongly prefer to avoid requiring users to install packages themselves outside of Spack to use Spack, particularly for something so fundamental. So loading a spack installed version (if it exsits) would be ideal. Does that sound acceptable to you?

Reply:
> This PR does not stop providing the the _7zip package from spack to users,

I understand: there is still a `7zip` package.

However, if you remove the `_7zip` method here, how will spack ever be able to use `7zip` to unpack an archive?

> I'd strongly prefer to avoid requiring users to install packages themselves outside of Spack to use Spack, particularly for something so fundamental.

I agree and to be clear, keeping the `_7zip` method and offering it as an option for `.Z` files does not conflict with that goal.

Reply:
> However, if you remove the _7zip method here, how will spack ever be able to use 7zip to unpack an archive?
Sorry I thought I had conveyed that I agree with you in my initial comment, and that response was an attempt to clarify the need for this PR/how it changes the 7zip dep. For clarity I disagree with my initial implementation, agree with the above, and for `Z` compressed archives, we _should_ absolutely be falling back to 7zip as there is no other option that would not also put us in the same position. 

Is querying the installed packages (like `spack.store.db.query("7zip")[0].prefix.bin`) the best way to achieve this, or is there a more idiomatic way in Spack I'm unaware of?